### ISSUE TITLE
some question about sampling code

### ISSUE THREAD

**Issue by LiXinghui-666** [zju3dv/NeuralRecon-W#25]
Hi, I'd like to ask some details about sampling in code because I want to do similar octree sampling using kaolin source library in my own dataset. I used the results of the indoor MVS as the initialization input of the octree (which is much denser than sfm result), but in the experiment, I found that when using kaolin.render.spc.unbatched_raytrace:
```python
ray_index, pt_ids, depth_in_out = spc_render.unbatched_raytrace(
        octree,
        points,
        pyramid,
        prefix,
        rays_o_normalized,
        rays_d,
        level,
        return_depth=True,
        with_exit=with_exit,
    )
```
The following information is displayed.As a result, the octree sampling method fails.
```python
[WARNING] batch has 0 intersections!!
```

 I can verify my coordinate system and camera pose visually. I want to know how to solve this problem.
The following figure shows the visualized octree point and the rays input to kaolin.spc_render.

![2](https://user-images.githubusercontent.com/45110729/193814794-00999c9d-9464-4c7b-944a-3e8b31954253.PNG)!

**Comment by cv-lab-x** [comment]
> I used the results of the indoor MVS as the initialization input of the octree

did you solve this problem, i got the same problem with you in indoor @LiXinghui-666

**Comment by cv-lab-x** [comment]
@Burningdust21 looking forward to your reply, thanks

**Comment by Burningdust21** [comment]
Hi, The warning means the rays in that batch didn't intersect with any surface, but this doesn't necessarily mean ray tracing failed as some regions might be empty. But if this happens more than it should,  it might be caused by an earlier version of kaolin that can't correctly produce intersections within the normalization range(more in this  [issue](https://github.com/NVIDIAGameWorks/kaolin/issues/490) ). Just Make sure you're using the right version in kaolin as in [requirements.txt](https://github.com/zju3dv/NeuralRecon-W/blob/main/requirements.txt); or you can check out the aforementioned issue and see the latest progress.

**Comment by tb2-sy** [comment]
> Hi, The warning means the rays in that batch didn't intersect with any surface, but this doesn't necessarily mean ray tracing failed as some regions might be empty. But if this happens more than it should, it might be caused by an earlier version of kaolin that can't correctly produce intersections within the normalization range(more in this [issue](https://github.com/NVIDIAGameWorks/kaolin/issues/490) ). Just Make sure you're using the right version in kaolin as in [requirements.txt](https://github.com/zju3dv/NeuralRecon-W/blob/main/requirements.txt); or you can check out the aforementioned issue and see the latest progress.

Yes, you are right, but the requirements.txt does not mention the kaolin version......

**Comment by Burningdust21** [comment]
It's actually in [environment.yaml](https://github.com/zju3dv/NeuralRecon-W/blob/main/environment.yaml).

**Comment by tb2-sy** [comment]
> It's actually in [environment.yaml](https://github.com/zju3dv/NeuralRecon-W/blob/main/environment.yaml).

Thank you for your reply, I found it in environment.yaml, but after installing this kaolin, I still reported an error, "AttributeError: module 'kaolin._C.ops.spc' has no attribute 'morton_to_points_cuda'".

**Comment by Dawars** [comment]
This can happen when you build kaolin without cuda available.
You can force it to build with cuda using `FORCE_CUDA=1 IGNORE_TORCH_VER=1 python setup.py develop`
